### PR TITLE
DDF-3566 Updated OpenSearchFilterVisitor for handling changes to wild…

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Applies normalization to OpenSearchFilterVisitor for handling the change in UI wildcard characters. This allows for non-escaped `*` searches on Open Search sources.

#### Who is reviewing it? 
@garrettfreibott 
@jlcsmith 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@bdeining

#### How should this be tested? (List steps with links to updated documentation)
1. Create a OpenSearch source. Give it a name, and the default parameters are enough to verify.
2. Enable CXF logging `log:set DEBUG org.apache.cxf`
3. Query the source for `*`
4. Verify a log message similar to the following is output `https://localhost:8993/services/catalog/query?start=1&count=250&mt=300000&dn=admin&q=*&src=local`. Note: `q=*` is the part we are verifying.

#### Any background context you want to provide?
This normalization was already handled by other sources, and this just adds handling for the OpenSearch source.

#### What are the relevant tickets?
[DDF-3566](https://codice.atlassian.net/browse/DDF-3566)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by mojogitoverhere** [comment]
I think some unit tests would be helpful to prevent any regressions

File: catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchFilterVisitor.java [hunk 1]
```diff
@@ -380,6 +385,36 @@ public Object visit(Literal expression, Object data) {
...
+  protected String normalizePattern(
+      String pattern, String wildcard, String singleChar, String escapeChar) {
+    StringBuilder sb = new StringBuilder(pattern.length());
+    for (int i = 0; i < pattern.length(); i++) {
+      char c = pattern.charAt(i);
+      if (c == escapeChar.charAt(0)) {
+        if (i + 1 < pattern.length()) {
+          i++;
+          String next = Character.toString(pattern.charAt(i));
+          if ("*".equals(next) || "?".equals(next) || "\\".equals(next)) {
```
Comment (mackncheesiest) [inline catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchFilterVisitor.java]: Why not just perform character comparison here and skip the `Character.toString` call?
And how will `next` ever be equal to "\\" if it's only a single character?

Comment (troymohl) [reply catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchFilterVisitor.java]: Good catch. Fixed.

**Comment by troymohl** [comment]
@mojogitoverhere I added a unit test to verify that the substitution correctly occurs.

**Comment by bdeining** [comment]
heroing now

**Comment by troymohl** [comment]
Ran sonarlint manually and there are no issues
screen shot 2018-03-30 at 12 11 33 pm

**Pull request merged**